### PR TITLE
Support Sm2 signature algo

### PR DESCRIPTION
**Pull request by chengwenxi**

### CODE REVIEW

File: src/main/java/iservice/sdk/module/impl/SM2TxServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                                .setSequence(baseAccount.getSequence()))
+
+                // TODO : Configurable
+                .setFee(TxOuterClass.Fee.newBuilder().setGasLimit(200000).addAmount(CoinOuterClass.Coin.newBuilder().setAmount("10").setDenom("point"))).build();
+
+        TxOuterClass.SignDoc signdoc = TxOuterClass.SignDoc.newBuilder()
+                .setBodyBytes(txBody.toByteString())
+                .setAuthInfoBytes(ai.toByteString())
+                .setAccountNumber(baseAccount.getAccountNumber())
+                .setChainId("irita")
```
Comment (chengwenxi) [inline src/main/java/iservice/sdk/module/impl/SM2TxServiceImpl.java]: Get `chain-id`,`gaslimit`,`fees` from config

**Review by prolenking — APPROVED** [review]

File: service-sdk.iml [hunk 2]
```diff
@@ -0,0 +1,123 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module org.jetbrains.idea.maven.project.MavenProjectsManager.isMavenModule="true" type="JAVA_MODULE" version="4">
```
Comment (chengwenxi) [inline service-sdk.iml]: ignore jetbrains idea config

File: src/test/java/iservice/sdk/SuperTest.java [hunk 3]
```diff
@@ -14,18 +15,21 @@
...
         ServiceClient client = ServiceClientFactory.getInstance()
                 .setOptions(options)
                 .addListener(new TestConsumerListener())
                 .addListener(new TestProviderListener())
                 .getClient();
-        client.getKeyService().recoverKey("consumer", "123456", "potato below health analyst hurry arrange shift tent elevator syrup clever ladder adjust agree dentist pass best space behind badge enemy nothing twice nut", true, 0, "");
-        client.getKeyService().recoverKey("provider", "123456", "help such mushroom spell cream cattle cute brush crucial boat flat system oxygen apart sock leave position jaguar winner violin manage exchange scissors employ", true, 0, "");
+
+        String addr = client.getKeyService().recoverKey("consumer", "123456", "liquid addict dinosaur tumble model air mansion juice purpose inherit discover project solve apology crew crack engage dawn miss foam achieve broom tenant minute", true, 0, "");
+        System.out.printf(addr);
```
Comment (chengwenxi) [inline src/test/java/iservice/sdk/SuperTest.java]: check addr

File: src/main/java/iservice/sdk/module/impl/SM2TxServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    @Override
+    public TxOuterClass.Tx signTx(TxOuterClass.TxBody txBody, String name, String password, boolean offline) throws ServiceSDKException, IOException, CryptoException {
+        Key key = this.keyService.getKey(name, password);
+
+        Auth.BaseAccount baseAccount = this.authService.queryAccount(key.getAddress());
+        TxOuterClass.AuthInfo ai = TxOuterClass.AuthInfo.newBuilder()
+                .addSignerInfos(
+                        TxOuterClass.SignerInfo.newBuilder()
+                                //.setPublicKey(Crypto.PublicKey.newBuilder().setAnyPubkey(Any.newBuilder().setUnknownFields(ByteString.copyFrom(encodedPubkey))))
```
Comment (chengwenxi) [inline src/main/java/iservice/sdk/module/impl/SM2TxServiceImpl.java]: why no need pubkey?

**Pull request merged**